### ISSUE TITLE
Packaging shell scripts uses bash specific syntax while env is set for /bin/sh

### ISSUE THREAD

**Issue by amr-labib** [pact-foundation/pact-ruby-standalone#22]
The shell scripts under /packaging/ use the sh environment:
`#!/bin/sh`
However, the syntax is bash specific. This is fine for systems the have `/bin/sh` pointing to bash. This is not the case for many new linux systems (`/bin/sh` points to dash in new debian based destros). 
For example:
`if [[ $TARGET == /* ]]; then` 
is bash specific and it would never evaluate to true and will cause issues when trying to create relative symlinks to the shell scripts.

Wouldn't it be better to change it to:
`#!/bin/bash`
If the aim was to provide a POSIX based script, then the logic will need to change slightly to accommodate using `[ <cmd> ]` instead of `[[ <comd> ]]`

Thanks
Amr

**Comment by mefellows** [comment]
I'm in favour of the `/bin/sh` default POSIX script if we can possibly avoid bash-isms.

**Comment by bethesque** [comment]
@amr-labib I pretty much always use bash, so I'm unfamiliar with what won't work in /bin/sh. I'm very happy to accept a PR if you'd like to submit one.

**Comment by amr-labib** [comment]
I have submitted a PR for the issue related to this one. I happy to have a look later on through all scripts and check for any non-POSIX logic.

**Comment by mefellows** [comment]
Just revisiting this in relation too https://github.com/pact-foundation/pact-node/issues/83.

I think we should update the scripts to either a) remove bashisms or b) explicitly note that bash is required.

It has been the cause of confusion in a few places (notably, anybody who attempts to Alpine linux runs into this e.g. https://github.com/pact-foundation/pact-js/issues/296#issuecomment-487251576).

Marking as enhancement.

**Comment by bethesque** [comment]
We should be able to do everything we need in /bin/sh. I'm pretty sure there was a change made at some point to attempt this, but obviously we didn't get everything.

**Comment by YOU54F** [comment]
the `bash` dependency has now been removed since `v2.3.1` after #115 has been merged/released

If anyone has any issues, please feel free to raise a new issue.
